File: java/interface/p4.java
interface X{
	void fun();
	
	default void bun(){
		System.out.println("bun p");

	};


}


class Y implements X{
	
	public void fun(){
		System.out.println("fun child");
		
	}

	static void gun(){
		System.out.println("gun child");

	}


	public void bun(){
		System.out.println("bun child");

	};

}


class Test{
	public static void main(String args[]){
		Y obj = new Y();

		obj.fun();
		obj.gun();

		obj.bun();




	}
}

File: CPP/dynamicObject/p4.cpp
#include<iostream>
using namespace std;


class Test{

	int x = 2;

	public:


		Test(){
			cout<<"Constructor"<<endl;

		}

		Test(int a){
			x =a;
		}

		
		Test(Test &ref){
			cout<<"copy constr"<<endl;
		}

		

		void fun(){
			cout<<x<<endl;
		}

};



int main(){
	
	Test obj {obj}; //calling copy const on same object, if there was no copy const explicitly defined then it produces garbage value initialization


	obj.fun();

	Test obj2{};
	obj2.fun();


}
